library/Fission/Web/Auth.hs

```diff
@@ -1,96 +1,47 @@
 module Fission.Web.Auth
   ( Checks
-  , server
+  , authWithContext
   , basic
-  , user
-  , checkUser
   , mkAuth
+  , module Fission.Web.Auth.Class
+  , module Fission.Web.Auth.Types
   ) where
 
-import           Crypto.BCrypt
 import           Servant
-
 import           Database.Esqueleto
-import qualified Database.Persist as P
 
 import           Fission.Models
-import           Fission.Platform.Heroku.AddOn
+import           Fission.Platform.Heroku.Auth.Types as Heroku
 import           Fission.Prelude
+import           Fission.Web.Auth.Class
+import           Fission.Web.Auth.Types
 
-type Checks = '[BasicAuthCheck (Entity User), BasicAuthCheck ByteString]
+type Checks = '[BasicAuthCheck (Entity User), BasicAuthCheck Heroku.Auth]
 
 -- | Construct an authorization context
 mkAuth ::
-  ( MonadHerokuAddOn m
-  , MonadDB          inner
-  , MonadLogger      inner
+  ( MonadAuth (Entity User) m
```

Comment:
What's the reason for making these their own Monad? They're only used here & we weren't threading through `cfg` or anything

Reply:
It lets us fully mock them out in tests (and potentially local dev if we want to e.g. disable that with a flag)